### DISCUSSION TITLE
Inconsistencies at different places

### DISCUSSION THREAD

**Discussion by saadulkh** [typst/typst#743] [General]
I'm not sure but what are the reasons for the inconsistencies through out the language?
For example:
1. `grids` can accept `array` for the `column` parameter but a `page` cannot.
2. `rect` can accept `array` for the `inset` parameter but a `table` cannot.
There may be some other examples, but I had these in my mind right now. Is this by design? This causes inconsistency and makes you go through the documentation of every function to see if it supports something or face the errors.

**Comment by laurmaedje** [comment]
1. Implementation limitation
2. Oversight
The language is pretty big, some inconsistencies are bound the creep in although we try to hard to make everything consistent. :)
